src/server_manager/web_app/ui_components/app-root.js

```diff
@@ -561,15 +561,29 @@ export class AppRoot extends mixinBehaviors
    * @param {string} language
    * @param {string} direction
    */
-  setLanguage(language, direction) {
+  async setLanguage(language, direction) {
+    const out = new Promise((resolve, reject) => {
+      let successHandler, failureHandler;
+      successHandler = () => {
+        this.removeEventListener('app-localize-resources-error', failureHandler);
```

Comment:
Can you explain the event listeners here? It seems like just wrapping the setLanguage logic in a promise should be enough.

Reply:
Oh thanks. Yes, that clears it up!

Reply:
This is a bit hard to read. Can you provide a method that all it does is load the resources and return the completion promise?

Then the setLanguage can read almost like it was before, except that it will be async and have an `await loadResources()`.

I'm not sure about the name. Perhaps loadResourcesAsync? loadResourcesPromise?

Reply:
Fixed after discussion online in 9d11943e51379020fda0b9357457e185e844dfc3